pymongo/auth.py

```diff
@@ -557,15 +573,15 @@ def speculate_command(self):
 
 
 class _X509Context(_AuthContext):
-    def speculate_command(self):
+    def speculate_command(self) -> SON[str, int | str]:
```

Comment:
`|` was introduced in Python 3.10 so we'll need to use `Union` instead. Actually it might be better to use `Dict[str, Any]` or `MutableMapping[str, Any]` here (unless it's important that this is actually SON).

Reply:
Got it, any objections to using MutableMapping and Mapping in general to standardize collection type annotations?

Reply:
+1 to using more general types where appropriate, including `Sequence`.